sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionalExpressions.scala

```diff
@@ -273,12 +274,26 @@ case class CaseWhenCodegen(
     val cases = branches.map { case (condExpr, valueExpr) =>
       val cond = condExpr.genCode(ctx)
       val res = valueExpr.genCode(ctx)
+      val (condFunc, condIsNull, condValue, resFunc, resIsNull, resValue ) =
+        if ((cond.code.length + res.code.length) > 1024 &&
```

Comment:
Actually my previous comment meant the sum of generated codes of all branches.

With latest commit, it is possible that the code size of each branch is in the range, but the sum of all branches still breaks 64KB limit.